app/src/main/java/org/schabi/newpipe/settings/preferencesearch/PreferenceSearchConfiguration.java

```diff
@@ -0,0 +1,154 @@
+package org.schabi.newpipe.settings.preferencesearch;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.text.TextUtils;
+
+import androidx.annotation.XmlRes;
+import androidx.preference.PreferenceCategory;
+import androidx.preference.PreferenceScreen;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BinaryOperator;
+import java.util.stream.Stream;
+
+public class PreferenceSearchConfiguration {
+    private final ArrayList<SearchIndexItem> itemsToIndex = new ArrayList<>();
+
+    private BinaryOperator<String> breadcrumbConcat =
+            (s1, s2) -> TextUtils.isEmpty(s1) ? s2 : (s1 + " > " + s2);
+
+    private PreferenceSearchFunction searcher = new PreferenceFuzzySearchFunction();
+
+    private final List<String> parserIgnoreElements = Arrays.asList(
+            PreferenceCategory.class.getSimpleName());
+    private final List<String> parserContainerElements = Arrays.asList(
+            PreferenceCategory.class.getSimpleName(),
+            PreferenceScreen.class.getSimpleName());
+
+
+    public void setBreadcrumbConcat(final BinaryOperator<String> breadcrumbConcat) {
+        this.breadcrumbConcat = Objects.requireNonNull(breadcrumbConcat);
+    }
+
+    public void setSearcher(final PreferenceSearchFunction searcher) {
+        this.searcher = Objects.requireNonNull(searcher);
+    }
+
+    /**
+     * Adds a new file to the index.
+     *
+     * @param resId The preference file to index
+     * @return SearchIndexItem
+     */
+    public SearchIndexItem index(@XmlRes final int resId) {
```

Comment:
I'd rename this to `addToIndex`. Also, the return value seems unused, remove it if that's the case